### DISCUSSION TITLE
How to query package information in xmake.lua?

### DISCUSSION THREAD

**Discussion by shaqtsui** [xmake-io/xmake#3466] [Q&A]
I would like to zip target with it's dependencies(dlls) in xmake.lua.
I read out the documents, but no lucky. 
Can I know is there an api can do this?

**Comment by shaqtsui** [comment]
I fingered a code snippet: 
```
import("core.project.project")
local pkg = project.required_package("bar")
pkg:get("installdir")
```

Pls advise is this best practice?

**Comment by waruqi** [comment]
```
add_requires("zlib")
target("xxx")
    on_load(function (target)
        target:pkg("zlib"):installdir()
    end)
    add_packages("zlib")
```

**Reply by shaqtsui** [reply]
If I need to do the zip inside a seprate task, then:
```
import("core.project.project")
project.target("foo"):pkg("bar"):installdir()
```
?

**Reply by waruqi** [reply]
??

**Reply by waruqi** [reply]
why not use `xrepo` command? xrepo info, xrepo export ...

**Reply by shaqtsui** [reply]
As I need that task to be in my xmake.lua. 
So that other people can invoke this dedicated task to zip target with dll dependencies.

**Reply by shaqtsui** [reply]
Need to hightlight that, I am working on target, not package. So I think I cannot use expor here?

**Reply by waruqi** [reply]
os.exec + `xrepo export -o` + zip in task

**Reply by shaqtsui** [reply]
Great, thanks
